File: mpp-proiect-repository-cristinahendre/lab8/Java/lab6/modul/NetworkModule/src/main/java/protoProtocol/ProtoUtils.java
package protoProtocol;

import Domain.CazCaritabil;
import Domain.Donatie;
import Domain.Donator;
import Domain.Voluntar;
import dto.DonatorDTO;
import dto.VoluntarDTO;

import java.util.ArrayList;
import java.util.List;


public class ProtoUtils {
    public static protoProtocol.ChatProtobufs.ChatRequest createLoginRequest(VoluntarDTO user){
        ChatProtobufs.Voluntar userDTO= ChatProtobufs.Voluntar.newBuilder().setEmail(user.getEmail()).setPasswd(user.getPasswd()).build();
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest.newBuilder().setType(protoProtocol.ChatProtobufs.ChatRequest.Type.Login)
               .addVoluntar(userDTO).build();
        return request;
    }


    public static protoProtocol.ChatProtobufs.ChatRequest GetNumeDonRequest(DonatorDTO user){
        ChatProtobufs.Donator userDTO= ChatProtobufs.Donator.newBuilder()
                .setNume(user.getNume()).build();
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest
                .newBuilder().setType(ChatProtobufs.ChatRequest.Type.SearchDonator)
                .setDonator(userDTO).build();
        return request;
    }



    public static protoProtocol.ChatProtobufs.ChatRequest SaveDonatieRequest(Donatie d){
        ChatProtobufs.Donatie dto= ChatProtobufs.Donatie.newBuilder()
                .setCaz(ChatProtobufs.CazCaritabil.newBuilder().
                        setId(d.getCaz().getId()).setNume(d.getCaz().getNume())
                        .setSuma(d.getCaz().getSuma_donata()))
                        .setDon(ChatProtobufs.Donator.newBuilder().
                                setId(d.getDonator().getId()).setNume(d.getDonator().getNume())
                                .setPrenume(d.getDonator().getPrenume())
                                .setAdresa(d.getDonator().getAdresa()).setNr(d.getDonator().getNrTelefon())).
                        setSuma(d.getSuma_donata()).build();
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest
                .newBuilder().setType(ChatProtobufs.ChatRequest.Type.SaveDonatie)
                .setDonatie(dto).build();
        return request;
    }


    public static protoProtocol.ChatProtobufs.ChatRequest SaveDonatorRequest(DonatorDTO d){
        ChatProtobufs.Donator dto= ChatProtobufs.Donator.newBuilder()
                .setNume(d.getNume()).setPrenume(d.getPrenume())
                .setAdresa(d.getAdresa()).setNr(d.getTel()).build();

        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest
                .newBuilder().setType(ChatProtobufs.ChatRequest.Type.SaveDonator)
                .setDonator(dto).build();
        return request;
    }


    public  static  Donatie getDonatie(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Donatie d =request.getDonatie();
        ChatProtobufs.CazCaritabil caz=d.getCaz();
        CazCaritabil caz1=new CazCaritabil(caz.getNume());
        caz1.setSuma_donata(caz.getSuma());
        caz1.setId(caz.getId());

        ChatProtobufs.Donator donator =d.getDon();
        Donator don1=new Donator(donator.getNume(), donator.getPrenume(), donator.getAdresa(),
                (long) donator.getNr());
        don1.setId(donator.getId());
        Donatie don =new Donatie(don1,caz1,d.getSuma());
        don.setId(d.getId());

        return don;
    }

    public static Donator getDonatorfromDTO(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Donator don=request.getDonator();
        Donator nou =new Donator(don.getNume(),don.getPrenume(),don.getAdresa(),
                (long) don.getNr());
        return nou;
    }

    public  static  Donatie getDonatie(ChatProtobufs.ChatResponse request){
        ChatProtobufs.Donatie d =request.getDonatie();
        ChatProtobufs.CazCaritabil caz=d.getCaz();
        CazCaritabil caz1=new CazCaritabil(caz.getNume());
        caz1.setSuma_donata(caz.getSuma());
        caz1.setId(caz.getId());

        ChatProtobufs.Donator donator =d.getDon();
        Donator don1=new Donator(donator.getNume(), donator.getPrenume(), donator.getAdresa(),
                (long) donator.getNr());
        don1.setId(donator.getId());
        Donatie don =new Donatie(don1,caz1,d.getSuma());
        don.setId(d.getId());

        return don;
    }




    public static protoProtocol.ChatProtobufs.ChatRequest GetDateDonRequest(DonatorDTO user){
        ChatProtobufs.Donator userDTO= ChatProtobufs.Donator.newBuilder()
                .setNume(user.getNume()).setPrenume(user.getPrenume())
                .setAdresa(user.getAdresa()).setNr(user.getTel()).build();
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest
                .newBuilder().setType(ChatProtobufs.ChatRequest.Type.GetDateDonator)
                .setDonator(userDTO).build();
        return request;
    }


    public static protoProtocol.ChatProtobufs.ChatRequest createLogoutRequest(Voluntar user){
        ChatProtobufs.Voluntar userDTO= ChatProtobufs.Voluntar.newBuilder()
                .setId(user.getId()).setNume(user.getNume())
                .setPrenume(user.getPrenume()).setEmail(user.getEmail())
                .setPasswd(user.getParola()).build();
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs
                .ChatRequest.newBuilder().setType(protoProtocol.ChatProtobufs.ChatRequest
                        .Type.Logout).addVoluntar(userDTO).build();

        return request;

    }

    public static protoProtocol.ChatProtobufs.ChatRequest createSendMesssageRequest(Voluntar message){
//        protoProtocol.ChatProtobufs.Message messageDTO= protoProtocol.ChatProtobufs.Message.newBuilder().
//                setSenderId(message.getSender().getId())
//                .setReceiverId(message.getReceiver().getId())
//                .setText(message.getText()).build();
//        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest.newBuilder()
//                .setType(protoProtocol.ChatProtobufs.ChatRequest.Type.SendMessage)
//                .setMessage(messageDTO).build();
//        return request;
        return null;
    }

    public static protoProtocol.ChatProtobufs.ChatRequest createLoggedFriendsRequest(Voluntar user){
//        protoProtocol.ChatProtobufs.User userDTO= protoProtocol.ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest.newBuilder()
//                .setType(protoProtocol.ChatProtobufs.ChatRequest.Type.GetLoggedFriends)
//                .setUser(userDTO).build();
//        return request;
        return null;
    }


    public static protoProtocol.ChatProtobufs.ChatResponse createOkResponse(Voluntar vol){
        protoProtocol.ChatProtobufs.ChatResponse response= ChatProtobufs.ChatResponse.newBuilder()
                .addVoluntar(0, ChatProtobufs.Voluntar.newBuilder().setId(vol.getId()).setNume(vol.getNume())
                .setEmail(vol.getEmail()).setPrenume(vol.getPrenume()).setPasswd(vol.getParola()))
                        .setType(ChatProtobufs.ChatResponse.Type.Ok).build();
        return response;
    }


    public static protoProtocol.ChatProtobufs.ChatResponse getDateDonResponse(Donator vol){
        protoProtocol.ChatProtobufs.ChatResponse response= ChatProtobufs.ChatResponse.newBuilder()
               .addDonator(0,ChatProtobufs.Donator.newBuilder().setId(vol.getId()).setNume(vol.getNume())
                       .setPrenume(vol.getPrenume()).setAdresa(vol.getAdresa())
                       .setNr(vol.getNrTelefon()))
                .setType(ChatProtobufs.ChatResponse.Type.GetDonator).build();
        return response;
    }



    public static ChatProtobufs.ChatResponse createLogoutResponse(){
        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.Logout).build();
        return response;
    }

    public static ChatProtobufs.ChatResponse createRefreshResponse(Donatie d){
        ChatProtobufs.Donatie userDTO= ChatProtobufs.Donatie.newBuilder()
                .setCaz(ChatProtobufs.CazCaritabil.newBuilder().
                        setId(d.getCaz().getId()).setNume(d.getCaz().getNume())
                        .setSuma(d.getCaz().getSuma_donata()))
                .setDon(ChatProtobufs.Donator.newBuilder().
                        setId(d.getDonator().getId()).setNume(d.getDonator().getNume())
                        .setPrenume(d.getDonator().getPrenume())
                        .setAdresa(d.getDonator().getAdresa()).setNr(d.getDonator().getNrTelefon())).
                        setSuma(d.getSuma_donata()).build();

        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.Refresh)
                .setDonatie(userDTO).build();
        return response;
    }


    public static ChatProtobufs.ChatResponse createSaveDonatieResponse(){
        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.NewDonatie).build();
        return response;
    }

    public static ChatProtobufs.ChatResponse createSaveDonatorResponse(){
        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.NewDonator).build();
        return response;
    }

    public static protoProtocol.ChatProtobufs.ChatResponse createErrorResponse(String text){
        protoProtocol.ChatProtobufs.ChatResponse response= protoProtocol.ChatProtobufs.ChatResponse.newBuilder()
                .setType(protoProtocol.ChatProtobufs.ChatResponse.Type.Error)
                .setError(text).build();
        return response;
    }

    public static protoProtocol.ChatProtobufs.ChatResponse createFriendLoggedInResponse(Voluntar user){
//        protoProtocol.ChatProtobufs.User userDTO= protoProtocol.ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//
//        protoProtocol.ChatProtobufs.ChatResponse response=protoProtocol.ChatProtobufs.ChatResponse.newBuilder()
//                .setType(protoProtocol.ChatProtobufs.ChatResponse.Type.FriendLoggedIn)
//                .setUser(userDTO).build();
//        return response;
        return null;
    }




    public static String getError(protoProtocol.ChatProtobufs.ChatResponse response){
        String errorMessage=response.getError();
        return errorMessage;
    }

    public static VoluntarDTO getVoluntar(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Voluntar vol =request.getVoluntar(0);
        VoluntarDTO dto=new VoluntarDTO(vol.getEmail(), vol.getPasswd());

        return dto;
    }

    public static Voluntar getVoluntarBun(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Voluntar vol =request.getVoluntar(0);
        Voluntar v=new Voluntar(vol.getNume(),vol.getPrenume(),vol.getEmail(),
                vol.getPasswd());
        v.setId(vol.getId());

        return v;
    }



    public static DonatorDTO getDonator(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Donator vol =request.getDonator();
        DonatorDTO dto=new DonatorDTO(vol.getNume());

        return dto;
    }

    public static DonatorDTO getDTODonator(ChatProtobufs.ChatRequest request){
        ChatProtobufs.Donator vol =request.getDonator();
        DonatorDTO dto=new DonatorDTO(vol.getNume(),vol.getPrenume(),vol.getAdresa(),
                (long) vol.getNr());

        return dto;
    }


    public static ChatProtobufs.ChatRequest createGetCazuriRequest() {
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol
                .ChatProtobufs.ChatRequest.newBuilder().setType(ChatProtobufs
                        .ChatRequest.Type.GetCazuri)
               .build();
        return request;
    }

    public static ChatProtobufs.ChatResponse createNullDonatorResp() {
        ChatProtobufs.ChatResponse request= ChatProtobufs.ChatResponse
                .newBuilder().setType(ChatProtobufs
                        .ChatResponse.Type.Unknown)

                .build();
        return request;
    }

    public static ChatProtobufs.ChatRequest createGetDonatoriAll() {
        protoProtocol.ChatProtobufs.ChatRequest request= protoProtocol.ChatProtobufs.ChatRequest.newBuilder()
                .setType(ChatProtobufs.ChatRequest.Type.GetDonatori)
                .build();
        return request;
    }



    public static protoProtocol.ChatProtobufs.ChatResponse GetCazuriResponse(Iterable<CazCaritabil> cazuri){
       List<ChatProtobufs.CazCaritabil> nou=new ArrayList<>();
        for(CazCaritabil c:cazuri){
            ChatProtobufs.CazCaritabil caz=ChatProtobufs.CazCaritabil.newBuilder().
                    setId(c.getId()).setNume(c.getNume()).setSuma(c.getSuma_donata()).build();
            nou.add(caz);
        }
        ChatProtobufs.ChatResponse.Builder response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.GetCazuri)
                .addAllCazuri(nou);


        return response.build();
    }



    public static protoProtocol.ChatProtobufs.ChatResponse GetDonatoriResponse(Iterable<Donator> donators){
        List<ChatProtobufs.Donator> nou=new ArrayList<>();
        for(Donator c:donators){
            ChatProtobufs.Donator caz=ChatProtobufs.Donator.newBuilder().
                    setId(c.getId()).setNume(c.getNume()).setPrenume(c.getPrenume())
                    .setAdresa(c.getAdresa()).setNr(c.getNrTelefon())
                    .build();
            nou.add(caz);
        }
        ChatProtobufs.ChatResponse.Builder response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.GetDonatori)
                .addAllDonator(nou);


        return response.build();
    }



    public static protoProtocol.ChatProtobufs.ChatResponse GetNumeDonResponse(Iterable<Donator> donators){
        List<ChatProtobufs.Donator> nou=new ArrayList<>();
        for(Donator c:donators){
            ChatProtobufs.Donator caz=ChatProtobufs.Donator.newBuilder().
                    setId(c.getId()).setNume(c.getNume()).setPrenume(c.getPrenume())
                    .setAdresa(c.getAdresa()).setNr(c.getNrTelefon())
                    .build();
            nou.add(caz);
        }
        ChatProtobufs.ChatResponse.Builder response=ChatProtobufs.ChatResponse.newBuilder()
                .setType(ChatProtobufs.ChatResponse.Type.GetNumeDon)
                .addAllDonator(nou);


        return response.build();
    }



    public static Iterable<CazCaritabil> getCazuriProto(ChatProtobufs.ChatResponse response){

        List<CazCaritabil> cazs=new ArrayList<>();
        int i=0;
        while(i!=response.getCazuriCount()){
            ChatProtobufs.CazCaritabil userDTO=response.getCazuri(i);
            CazCaritabil user=new CazCaritabil(userDTO.getNume());
            user.setSuma_donata(userDTO.getSuma());
            user.setId(userDTO.getId());
            cazs.add(user);
            i++;
        }
        return cazs;
    }


    public static Iterable<Donator> getDonatoriProto(ChatProtobufs.ChatResponse response){

        List<Donator> cazs=new ArrayList<>();
        int i=0;
        while(i!=response.getDonatorCount()){
            ChatProtobufs.Donator userDTO=response.getDonator(i);
            Donator d=new Donator(userDTO.getNume(), userDTO.getPrenume(), userDTO.getAdresa(),
                    (long) userDTO.getNr());
            d.setId(userDTO.getId());
            cazs.add(d);
            i++;
        }
        return cazs;
    }
}

File: mpp-proiect-repository-cristinahendre/lab8/CSharp/Lab5/protobuf/ProtoUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Cerinta1.Domain;
using Chat.Protocol;
using Network;
using proto=Chat.Protocol;

namespace protobuf
{
    static class ProtoUtils
    {
        public static ChatRequest createLoginRequest(VoluntarDTO user)
        {
            //new proto.User {Id = user.Id, Passwd = user.Password};
            proto.Voluntar userDTO = new proto.Voluntar { Email = user.Email, Passwd = user.Passwd };
            ChatRequest request = new ChatRequest { Type = ChatRequest.Types.Type.Login};
            request.Voluntar.Add(userDTO);

            return request;
        }

        public static ChatRequest createLogoutRequest(Cerinta1.Domain.Voluntar user)
        {
            proto.Voluntar userDTO = new proto.Voluntar { Id = user.Id, Nume=user.Nume,
                                      Email=user.Email, Passwd=user.Parola};
            ChatRequest request = new ChatRequest { Type = ChatRequest.Types.Type.Logout };
            request.Voluntar.Add(userDTO);
            return request;
        }
        public static ChatRequest GetNumeDonRequest(DonatorDTO message)
        {
            proto.Donator messageDTO = new proto.Donator
            {
                Nume = message.Nume
            };

            ChatRequest request = new ChatRequest { Type = ChatRequest.Types.Type.SearchDonator, Donator = messageDTO };
            return request;
        }


        public static ChatRequest SaveDonatieRequest(Cerinta1.Domain.Donatie don)
        {
            proto.CazCaritabil caz = new proto.CazCaritabil
            {
                Id = don.Caz.Id,
                Nume = don.Caz.Nume,
                Suma = don.Caz.Suma_donata
            };

            proto.Donator donator = new proto.Donator
            {
                Id = don.Donator.Id,
                Nume = don.Donator.Nume,
                Prenume = don.Donator.Prenume,
                Adresa = don.Donator.Adresa,
                Nr = don.Donator.NrTelefon
            };
            proto.Donatie userDTO = new proto.Donatie {
                Don=donator,
                Caz=caz,
                Suma=don.Suma_donata

            };
            ChatRequest request = new ChatRequest
            {
                Type = ChatRequest.Types.Type.SaveDonatie,
                Donatie = userDTO
            };
            return request;
        }


        public static ChatResponse createOkResponse()
        {
            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.Ok };
            return response;
        }


        public static ChatResponse createErrorResponse(String text)
        {
            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.Error,
                Error = text
            };
            return response;
        }

        public static ChatResponse createOkResponse(Cerinta1.Domain.Voluntar user)
        {
            proto.Voluntar dto = new proto.Voluntar { 
                Id = user.Id,
                Nume = user.Nume,
                Prenume=user.Prenume,
                Email=user.Email,
                Passwd=user.Parola
            };
            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.Ok
                };
            response.Voluntar.Add(dto);
            return response;
        }


        public static ChatResponse getDateDonResponse(Cerinta1.Domain.Donator user)
        {
            proto.Donator userDTO = new proto.Donator { Id = user.Id ,
                Nume=user.Nume,
                Prenume=user.Prenume,
                Adresa=user.Adresa,
                Nr=user.NrTelefon
            };
            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.GetDonator
            };
            response.Donator.Add(userDTO);
            return response;
        }


        public static ChatResponse createLogoutResponse()
        {
            

            ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.Logout };
            return response;
        }

        public static ChatRequest createGetCazuriRequest()
        {


            ChatRequest response = new ChatRequest { 
                Type = ChatRequest.Types.Type.GetCazuri
            };
            return response;
        }

        public static ChatRequest createGetDonatoriAll()
        {


            ChatRequest response = new ChatRequest
            {
                Type = ChatRequest.Types.Type.GetDonatori
            };
            return response;
        }

        public static ChatResponse GetCazuriResponse(IEnumerable<Cerinta1.Domain.CazCaritabil> cazuri)
        {
            IList<proto.CazCaritabil> cazs = new List<proto.CazCaritabil>();
            foreach(Cerinta1.Domain.CazCaritabil c in cazuri)
            {
                proto.CazCaritabil ca = new proto.CazCaritabil
                {
                    Id = c.Id,
                    Nume = c.Nume,
                    Suma = c.Suma_donata
                };

                cazs.Add(ca);


            }

            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.GetCazuri,
                
            };
            response.Cazuri.Add(cazs);

            return response;

        }



        public static ChatResponse GetDonatoriResponse(IEnumerable<Cerinta1.Domain.Donator> cazuri)
        {
            IList<proto.Donator> cazs = new List<proto.Donator>();
            foreach (Cerinta1.Domain.Donator c in cazuri)
            {
                proto.Donator ca = new proto.Donator
                {
                    Id = c.Id,
                    Nume = c.Nume,
                    Prenume=c.Prenume,
                    Adresa=c.Adresa,
                    Nr=c.NrTelefon
                };

                cazs.Add(ca);


            }

            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.GetDonatori

            };
            response.Donator.Add(cazs);

            return response;

        }


        public static ChatResponse createRefreshResponse(Cerinta1.Domain.Donatie d)
        {
            proto.CazCaritabil caz = new proto.CazCaritabil
            {
                Id = d.Caz.Id,
                Nume = d.Caz.Nume,
                Suma = d.Caz.Suma_donata
            };

            proto.Donator donator= new proto.Donator
            {
                Id=d.Donator.Id,
                Nume=d.Donator.Nume,
                Prenume=d.Donator.Prenume,
                Adresa=d.Donator.Adresa,
                Nr=d.Donator.NrTelefon
            };


            proto.Donatie dona = new proto.Donatie
            {
                Caz = caz,
                Don = donator,
                Suma = d.Suma_donata
            };


            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.Refresh,
                Donatie=dona
            };
         
            return response;
        }

        public static String getError(ChatResponse response)
        {
            String errorMessage = response.Error;
            return errorMessage;
        }

        public static ChatResponse createSaveDonatieResponse()
        {

            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.NewDonatie
            };

            return response;
        }


        public static ChatResponse createSaveDonatorResponse()
        {

            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.NewDonator
            };

            return response;
        }

        public static VoluntarDTO getVoluntar(ChatRequest resp)
        {
            proto.Voluntar  vol = resp.Voluntar[0];
            VoluntarDTO dto = new VoluntarDTO(vol.Email, vol.Passwd);
            return dto;
        }

        public static Cerinta1.Domain.Voluntar getVoluntarBun(ChatRequest req)
        {
            proto.Voluntar vol = req.Voluntar[0];
            Cerinta1.Domain.Voluntar v = new Cerinta1.Domain.Voluntar(vol.Nume, vol.Prenume, vol.Email, vol.Passwd);
            v.Id = vol.Id;
            return v;


        }


        public static ChatRequest SaveDonatorRequest(DonatorDTO d)
        {
            proto.Donator dona = new proto.Donator
            {
                Nume = d.Nume,
                Prenume = d.Pre,
                Adresa = d.Adresa,
                Nr = d.Tel
            };

            ChatRequest request = new ChatRequest
            {
                Type = ChatRequest.Types.Type.SaveDonator,
                Donator = dona
            };
            return request;

        }

        public static ChatResponse createNullDonatorResp()
        {

            ChatResponse request = new ChatResponse
            {
                Type = ChatResponse.Types.Type.Unknown
              
            };
            return request;
        }

        public static Cerinta1.Domain.Donator getDonatorfromDTO(ChatRequest req)
        {
            proto.Donator dona = req.Donator;
            Cerinta1.Domain.Donator nou = new Cerinta1.Domain.Donator(dona.Nume, dona.Prenume,
                dona.Adresa, (long)dona.Nr);

            return nou;

        }




        public static DonatorDTO getDonator(ChatRequest response)
        {
            proto.Donator d = response.Donator;
            DonatorDTO don = new DonatorDTO(d.Nume);
            return don;
        }

        public static DonatorDTO getDTODonator(ChatRequest req)
        {
            proto.Donator d = req.Donator;
            DonatorDTO don = new DonatorDTO(d.Nume,d.Prenume,d.Adresa, (long)d.Nr);
            return don;

        }

        public static Cerinta1.Domain.Donatie getDonatie(ChatRequest request)
        {
            proto.Donatie d = request.Donatie;
            proto.CazCaritabil caz = d.Caz;
            Cerinta1.Domain.CazCaritabil c = new Cerinta1.Domain.CazCaritabil(caz.Nume,
                caz.Suma);
            c.Id = caz.Id;

            proto.Donator don = d.Don;
            Cerinta1.Domain.Donator d1= new Cerinta1.Domain.Donator(don.Nume, don.Prenume,
                don.Adresa, (long)don.Nr);
            d1.Id = don.Id;

            Cerinta1.Domain.Donatie donatia = new Cerinta1.Domain.Donatie(c, d.Suma, d1);
            return donatia;
        }


        public static Cerinta1.Domain.Donatie getDonatie(ChatResponse request)
        {
            proto.Donatie d = request.Donatie;
            proto.CazCaritabil caz = d.Caz;
            Cerinta1.Domain.CazCaritabil c = new Cerinta1.Domain.CazCaritabil(caz.Nume,
                caz.Suma);
            c.Id = caz.Id;

            proto.Donator don = d.Don;
            Cerinta1.Domain.Donator d1 = new Cerinta1.Domain.Donator(don.Nume, don.Prenume,
                don.Adresa, (long)don.Nr);
            d1.Id = don.Id;

            Cerinta1.Domain.Donatie donatia = new Cerinta1.Domain.Donatie(c, d.Suma, d1);
            return donatia;
        }


        public static ChatRequest GetDateDonRequest(DonatorDTO dona)
        {
            proto.Donator sender = new proto.Donator
            {
                Nume = dona.Nume,
                Prenume = dona.Pre,
                Adresa = dona.Adresa,
                Nr = dona.Tel
            };

            ChatRequest request = new ChatRequest
            {
                Type = ChatRequest.Types.Type.GetDateDonator,
                Donator = sender
            };
            return request;
        }



        public static ChatResponse GetNumeDonResponse(IEnumerable<Cerinta1.Domain.Donator> donators)
        {
            IList<proto.Donator> cazs = new List<proto.Donator>();
            foreach (Cerinta1.Domain.Donator c in donators)
            {
                proto.Donator ca = new proto.Donator
                {
                    Id = c.Id,
                    Nume = c.Nume,
                    Prenume = c.Prenume,
                    Adresa = c.Adresa,
                    Nr = c.NrTelefon
                };

                cazs.Add(ca);


            }

            ChatResponse response = new ChatResponse
            {
                Type = ChatResponse.Types.Type.GetNumeDon

            };
            response.Donator.Add(cazs);
            return response;
        }

        public static IEnumerable<Cerinta1.Domain.CazCaritabil> getCazuriProto(ChatResponse rep)
        {
            IList<Cerinta1.Domain.CazCaritabil> cazuri = new List<Cerinta1.Domain.CazCaritabil>();
            int i = 0;
            while(i!= rep.Cazuri.Count)
            {
                proto.CazCaritabil ca = rep.Cazuri[i];
                Cerinta1.Domain.
                    CazCaritabil c = new Cerinta1.Domain.CazCaritabil(ca.Nume, ca.Suma);
                c.Id = ca.Id;
                cazuri.Add(c);


                i++;
            }
            return cazuri;
        }


        public static IEnumerable<Cerinta1.Domain.Donator> getDonatoriProto(ChatResponse rep)
        {
            IList<Cerinta1.Domain.Donator> cazuri = new List<Cerinta1.Domain.Donator>();
            int i = 0;
            while (i != rep.Donator.Count)
            {
                proto.Donator ca = rep.Donator[i];
                Cerinta1.Domain.
                    Donator c = new Cerinta1.Domain.Donator(ca.Nume, ca.Prenume,ca.Adresa,
                    (long)ca.Nr);
                c.Id = ca.Id;
                cazuri.Add(c);


                i++;
            }
            return cazuri;
        }

    }
}
